cmd/flarectl/zone.go

```diff
@@ -196,3 +196,194 @@ func zoneRecords(c *cli.Context) {
 	}
 	makeTable(output, "ID", "Type", "Name", "Content", "Proxied", "TTL")
 }
+
+func zoneCustHostList(c *cli.Context) {
+	if err := checkEnv(); err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	var zone string
+	if c.String("zone") != "" {
+		zone = c.String("zone")
+	} else {
+		cli.ShowSubcommandHelp(c)
+		return
+	}
+	zoneID, err := api.ZoneIDByName(zone)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	customhostnamesResult, _, err := api.CustomHostnames(zoneID, 0, cloudflare.CustomHostname{})
```

Comment:
Need to wire up pagination (I recognize it's on your to-do list) as most customers have hundreds-to-tens-of-thousands of these by default.